pep-0496-environment-markers.rst

```diff
@@ -48,14 +50,32 @@ Environment Markers attempt to provide more flexibility in a list of
 requirements by allowing the developer to list requirements that are
 specific to a particular environment.
 
+Alternatives considered
+=======================
+
+For a while, OpenStack used seperate requirements files for different
+environments - eg, ``requirements-py2.txt`` and
+``requirements-py3.txt``. However, in most cases the two files were
+near duplicates, with only one or two lines different; but maintaining
+two seperate files was more than twice as much work, as it required
+manual checking to keep the files in sync.
+
+Keeping all the requirements together in one place, with markers
+indicating which to install, has been easier to maintain.
+
 Examples
 ========
 
-Here are some examples of such markers inside a requirements.txt::
+Here is an example of such markers inside a requirements.txt
+file. This will install ``pywin32`` on Windows; ``unittest2`` on
+python2.4 or python2.5; ``backports.ssl_match_hostname`` on the
+versions where ssl_match_hostname is not already implemented; and
+``hawkey`` only on linux::
 
    pywin32 >=1.0 ; sys_platform == 'win32'
    unittest2 >=2.0,<3.0 ; python_version == '2.4' or python_version == '2.5'
-   backports.ssl_match_hostname >= 3.4 ; python_version < '2.7.9' or (python_version >= '3.0' and python_version < '3.4')
+   backports.ssl_match_hostname >= 3.4 ; python_version < '3.4' and python_version != '2.7.10'
```

Comment:
So this example will be broken when 2.7.11 comes out. I think thats undesirable.